chore: Replacing Juju bundles with Terraform modules in Deploy SD-Core Standalone and Integrate SD-Core with Canonical Observability Stack (#192)
canonical/charmed-5g: docs/how-to/integrate_sdcore_with_observability.md

### Adding COS to an existing SD-Core deployment

Go to a directory from which SD-Core was deployed (the one containing Terraform's `.tfstate` file).
Edit the `.tfvars` and add following line(s) to it:

```console
deploy_cos = true
cos_model_name = "<YOUR_CUSTOM_COS_MODEL_NAME>" (Optional. Defaults to `cos-lite`.)
cos_configuration_config = {} (Optional. Allows customization of the `COS Configuration` application.)
```

Apply the changes:

```console
terraform apply -var-file="<YOUR_TFVARS_FILE>" -auto-approve
```

Monitor the status of the deployment:

```console
juju switch <YOUR_CUSTOM_COS_MODEL_NAME>
watch -n 1 -c juju status --color --relations
```

The deployment is ready when all the charms are in the `Active/Idle` state.

## Accessing the 5G Network Overview Grafana dashboard

Retrieve the Grafana URL and admin password:

```console
juju switch cos-lite
juju run grafana/leader get-admin-password
```

You should see the output similar to the following:

```console
Running operation 1 with 1 task
  - task 2 on unit-grafana-0

Waiting for task 2...
admin-password: c72uEq8FyGRo
url: http://10.201.0.51/cos-lite-grafana
```

```{note}

> Due to a bug in the Traefik charm, the URL returned by the command shown above, shows invalid `http` protocol.

Reviewer: ```suggestion
Due to a bug in the Traefik charm, the URL returned incorrectly shows `http` as the protocol.
```

Author: Good point. The old version suggested invalidity of the `http` as such.
I'd keep `by the command above,` tho. 
```Due to a bug in the Traefik charm, the URL returned by the command above, incorrectly shows `http` as the protocol.
```
WDYT?

Reviewer: Other than the `,` in between the words above and incorrectly, looks good

Author: I've changed the note completely, because now both `http` and `https` work :)

cos_configuration_config = {} (Optional. Allows customization of the `COS Configuration` application.)
```

```{note}
If you have already created the `.tfvars` file, to customize the deployment of SD-Core, you should edit the existing file rather than create a new one.
```

Proceed with the deployment.

(option-2)=
## Integrating COS with an existing SD-Core deployment

This option allows deploying COS and integrating it with SD-Core as a Day 2 operation.

### Pre-requisites

- A Kubernetes cluster capable of handling the load from both SD-Core and COS
- Any [Charmed 5G Terraform module][Charmed 5G Terraform modules] deployed

### Adding COS to an existing SD-Core deployment

Go to a directory from which SD-Core was deployed (the one containing Terraform's `.tfstate` file).
Edit the `.tfvars` and add following line(s) to it:

```console
deploy_cos = true
cos_model_name = "YOUR_CUSTOM_COS_MODEL_NAME" (Optional. Defaults to `cos-lite`.)
cos_configuration_config = {} (Optional. Allows customization of the `COS Configuration` application.)
```

Apply the changes:

```console
terraform apply -var-file="<YOUR_TFVARS_FILE>" -auto-approve

> ```

Reviewer 2: Here, I think we need add some comments to wait till all applications become Active/Idle state. Because the next stage directly starts with getting grafana admin password. If we the users immediately goes to the next stage, the action will fail.

Author: Added.

## Accessing the 5G Network Overview Grafana dashboard

Retrieve the Grafana URL and admin password:

```console
juju switch cos-lite
juju run grafana/leader get-admin-password
```

You should see the output similar to the following:

```console
Running operation 1 with 1 task
  - task 2 on unit-grafana-0

Waiting for task 2...
admin-password: c72uEq8FyGRo
url: http://10.201.0.51/cos-lite-grafana
```

```{note}

> Due to a bug in Traefik, the URL returned by the command shown above, shows invalid `http` protocol.

Reviewer 2: ```suggestion
Due to a bug in the Traefik, the URL returned by the command shown above which shows invalid `http` protocol.
```

Author: Updated.
